src/api/ReducerClient.js

```diff
@@ -0,0 +1,39 @@
+import BaseClient from "api/BaseClient";
+import { registerHistoryStackPushEvent } from "hooks/PreviousQuotesHistory";
+
+class ReducerClient extends BaseClient {
+  constructor(dispatch) {
+    super();
+    this.dispatch = dispatch;
+  }
+
+  updateState(json) {
+    // Register a history stack push event, to be picked up by our history
+    // hook when the state has been updated with this response.
+    // We do this here because we need the updated state in the history item
+    // which is pushed onto the stack.
+    registerHistoryStackPushEvent();
+
+    // Call dispatch Reducer action for setting quotes data in context state
+    this.dispatch({
+      type: "ajax/setQuotesData",
```

Comment:
```suggestion
      type: "ajax/setQuotesData",
```

Is this intended to be a generic reducer? If so maybe the dispatch type could be generic or it could be passed in?

Reply:
The reducer handles all state changes for the Previous Quotes view, not 100% sure how you mean this could be generic - Perhaps an example if you have time? Cheers!